frame/merkle-mountain-range/src/lib.rs

```diff
@@ -334,6 +334,31 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		mmr.generate_batch_proof(leaf_indices)
 	}
 
+	/// Generate a MMR proof for the given `leaf_indices` at the `best_known_block`.
+	///
+	/// Note this method can only be used from an off-chain context
+	/// (Offchain Worker or Runtime API call), since it requires
+	/// all the leaves to be present.
+	/// It may return an error or panic if used incorrectly.
+	pub fn generate_historical_batch_proof(
+		leaf_indices: Vec<LeafIndex>,
+		best_known_block: <T as frame_system::Config>::BlockNumber,
+	) -> Result<
+		(Vec<LeafOf<T, I>>, primitives::BatchProof<<T as Config<I>>::Hash>),
+		primitives::Error,
+	>
+	where
+		<T as frame_system::Config>::BlockNumber: Into<LeafIndex>,
+	{
+		let block_age = <frame_system::Pallet<T>>::block_number()
+			.saturating_sub(best_known_block)
+			.into();
+		let leaves_count = Self::mmr_leaves().saturating_sub(block_age);
+
+		let mmr: ModuleMmr<mmr::storage::OffchainStorage, T, I> = mmr::Mmr::new(leaves_count);
+		mmr.generate_batch_proof(leaf_indices)
```

Comment:
It won't panic if `leaves_count` is zero and `leaf_indices` is e.g. `[10]`, right?